lib/src/server/server.dart

```diff
@@ -100,14 +100,14 @@ class Server {
     new ServerHandler(lookupService, stream).handle();
   }
 
-  Future<Null> shutdown() {
+  Future<Null> shutdown() async {
     final done = _connections.map((connection) => connection.finish()).toList();
     if (_insecureServer != null) {
       done.add(_insecureServer.close());
     }
     if (_secureServer != null) {
       done.add(_secureServer.close());
     }
-    return Future.wait(done);
+    await Future.wait(done);
```

Comment:
Is this fixing a Dart2 type error or did you intend to change async ordering? The upcoming sync-async change should make the two behave identically as far as the latter goes. (If there is a non-obvious bugfix here, maybe add a test?).

Reply:
It's fixing a type error - Future.wait() returns a Future<List<T>>, and we claim to return a Future<Null>. Analyzer doesn't care about this, but the VM in checked mode does at runtime...

Async ordering shouldn't matter much here - we're shutting down the server.